runtime-native/src/wasm32.rs

```diff
@@ -0,0 +1,48 @@
+use futures::prelude::*;
+use futures::{future::BoxFuture, task::SpawnError};
+// use futures::compat::*;
+
+use std::io;
+use std::net::SocketAddr;
+use std::pin::Pin;
+
+use wasm_bindgen::prelude::*;
+use wasm_bindgen_futures::future_to_promise;
+
+/// The Native runtime.
+#[derive(Debug)]
+pub struct Native;
+
+impl runtime_raw::Runtime for Native {
+    fn spawn_boxed(&self, fut: BoxFuture<'static, ()>) -> Result<(), SpawnError> {
+        use futures01::future::Future;
+        let fut = fut
+            .unit_error()
+            .compat()
+            .map(|_| JsValue::undefined())
+            .map_err(|_| JsValue::undefined());
+        wasm_bindgen_futures::spawn_local(fut);
```

Comment:
https://rustwasm.github.io/wasm-bindgen/api/src/wasm_bindgen_futures/lib.rs.html#397-406 We shouldn't have to do the `map` and `map_err` now.

Reply:
ah yeah, good catch!